### PR TITLE
Simplify error handling

### PR DESCRIPTION
**Pull request by ArtskydJ**
I got confused reading #6, so I created this PR.

I tried to reduce the number of possible code paths.

Closes #6

### CODE REVIEW

File: index.mjs [hunk 1]
```diff
@@ -90,38 +89,39 @@ const sign_and_request = async({ sign, host, url, body }) => {
...
-	const [ failed_response, successful_response ] = await catchify(post(url, {
-		headers: {
-			...headers,
-			authorization,
-		},
-		body: body_string,
-	}))
+	let successful_response
+	try {
+		successful_response = await post(url, {
```
Comment (TehShrike) [inline index.mjs]: this makes me want to put `catchify` back 😂

File: index.mjs [hunk 2]
```diff
@@ -90,38 +89,39 @@ const sign_and_request = async({ sign, host, url, body }) => {
...
-		}
+			const error_element = drill_down_to_children(parse_xml_or_throw(await get_response_text(failed_response)), `ErrorResponse`, `Error`)[0]
+
+			const type = read_text_from_descendant(error_element, `Type`)
+			const code = read_text_from_descendant(error_element, `Code`)
+			const message = read_text_from_descendant(error_element, `Message`)
 
+			error_to_throw = new Error(`AWS error. type: "${ type }", code: "${ code }", message: "${ message }"`)
+			error_to_throw.code = code
+		} catch (_) {}
```
Comment (TehShrike) [inline index.mjs]: this try/catch block of error-generating-code where we want to swallow any errors that do happen would probably be a bit nicer in its own function

**Comment by TehShrike** [comment]
Yeah other than those thoughts I think your code paths here are probably better than #6

File: index.mjs (hunk 2 above)
Comment (ArtskydJ) [reply index.mjs]: done

File: index.mjs (hunk 1 above)
Comment (ArtskydJ) [reply index.mjs]: done... it improved this commit: a34b6ca66e4f7b069109899474070b2ed560728c
and probably the one after

**Comment by ArtskydJ** [comment]
🔄

**Review by TehShrike — APPROVED** [review]
yup yup lgtm

**Pull request merged**